### PR TITLE
Add loading state to ResultsCount

### PR DESCRIPTION
**Pull request by noobadmin**

### CODE REVIEW

File: src/App/ResultsCount.js [hunk 1]
```diff
@@ -26,7 +27,7 @@ function ResultsCount() {
 
   const defaultValue = process.env.REACT_APP_LIMIT || COUNT_OPTIONS[0]
 
-  return (
+  return count ? (
```
Comment (axelboc) [inline src/App/ResultsCount.js:30]: You should check for `=== undefined` in case `count` is indeed `0`. :wink:

**Comment by axelboc** [comment]
Just so I understand, what's the purpose of that call to `setCount()`?

**Comment by noobadmin** [comment]
The purpose is to show loading indicator in results count by undefining count. Here's where I've originally tried to put it: https://github.com/panosc-eu/searchui/blob/1c223a427ed0a23b81dde6114a29bd4a33a46df2/src/Explore/DocumentList.js#L33

**Comment by axelboc** [comment]
What about in:

```
useEffect(()=>{
  setCount(undefined);
}, [filters]);
```

You could put that in `ExplorePage` maybe.

**Comment by axelboc** [comment]
Sorry, `filters` and `query` don't seem to be stable. Ideally, the dependency should be the query URL passed to `useSWRImmutable`. This way, anytime it changes, you know SWR is making another request, which is when you want to reset the count.

**Comment by axelboc** [comment]
That being said, I reckon it would sufficient to reset the count only when changing the limit, in the `onChange` of the limit drop-down.

**Comment by noobadmin** [comment]
> That being said, I reckon it would sufficient to reset the count only when changing the limit, in the `onChange` of the limit drop-down.
Yes, it's indeed a bit awkward UI wise with just that "Spinner" now.

**Review by axelboc — APPROVED** [review]

**Pull request merged**